### ISSUE TITLE
The rule boolean-prop-naming does not check nested props.

### ISSUE THREAD

**Issue by zozoens31** [yannickcr/eslint-plugin-react#2041]
When using a shape prop, with a bool prop inside, it is not checked by the rule:

```javascript
static propTypes = 
  nest: PropTypes.shape({
    eggIsWhite: PropTypes.bool,  // No react/boolean-prop-naming warning here.
  }),
  thisIsChecked: PropTypes.bool,  // react/boolean-prop-naming warning.
}
```

**Comment by pawelnvk** [comment]
This should be fixed with https://github.com/yannickcr/eslint-plugin-react/pull/2234

**Comment by jzabala** [comment]
> This should be fixed with #2234

@ljharb should this one be closed then?

**Comment by ljharb** [comment]
Yep, thanks :-) closed by #2234.
